runtime/jcl/common/thread.cpp

```diff
@@ -326,18 +326,58 @@ Java_java_lang_Thread_getStackTraceImpl(JNIEnv *env, jobject rcv)
 	J9VMThread *currentThread = (J9VMThread*)env;
 	J9JavaVM *vm = currentThread->javaVM;
 	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	jobject result = NULL;
 
 	vmFuncs->internalEnterVMFromJNI(currentThread);
 	j9object_t receiverObject = J9_JNI_UNWRAP_REFERENCE(rcv);
 
-	/* Assume the thread is alive (guaranteed by java caller) */
-	J9VMThread *targetThread = J9VMJAVALANGTHREAD_THREADREF(currentThread, receiverObject);
+#if JAVA_SPEC_VERSION >= 19
+	BOOLEAN releaseInspector = TRUE;
+	if (IS_VIRTUAL_THREAD(currentThread, receiverObject)) {
+		omrthread_monitor_enter(vm->liveVirtualThreadListMutex);
+		j9object_t carrierThread = (j9object_t)J9VMJAVALANGVIRTUALTHREAD_CARRIERTHREAD(currentThread, receiverObject);
+		I_64 vthreadInspectorCount = J9OBJECT_I64_LOAD(currentThread, receiverObject, vm->virtualThreadInspectorCountOffset);
+
+		/* Ensure virtual thread is mounted and not during transition */
+		if ((NULL != carrierThread) && (vthreadInspectorCount >= 0)) {
+			J9OBJECT_I64_STORE(currentThread, receiverObject, vm->virtualThreadInspectorCountOffset, vthreadInspectorCount + 1);
+
+			/* Points receiver to carrierThread object */
+			receiverObject = carrierThread;
```

Comment:
After reconsidering, this may become an issue in the future if the code is further modified. We must set `releaseInspector` to true only when the counter has been incremented.

```
BOOLEAN releaseVThreadInspector = FALSE;
if (isVirtualThread) {
    if (counter incremented) { releaseVThreadInspector = TRUE; }
    else { goto release; }
} 

Run getStackTraceForThread for all threads

if (releaseVThreadInspector) { Decrement the counter }

release:
    vmFuncs->internalExitVMToJNI(currentThread);

```

Reply:
fixed